src/windows/models/files_model.py

```diff
@@ -385,34 +393,68 @@ def get_image_sequence_details(self, file_path):
                "%s%s.%s" % (full_base_name, str(x).rjust(digits, "0") if fixlen else str(x), extension)):
-                is_sequence = True
-                break
+                break  # found one!
         else:
-            is_sequence = False
-
-        if is_sequence and dirName not in self.ignore_image_sequence_paths:
-            log.info('Prompt user to import image sequence from {}'.format(dirName))
-            # Ignore this path (temporarily)
-            self.ignore_image_sequence_paths.append(dirName)
-
-            if not get_app().window.promptImageSequence(fileName):
-                # User said no, don't import as a sequence
-                return None
-
-            # Yes, import image sequence
-            parameters = {
-                "folder_path": dirName,
-                "base_name": base_name,
-                "fixlen": fixlen,
-                "digits": digits,
-                "extension": extension
-            }
-            return parameters
-
-        # We didn't discover an image sequence
-        return None
-
-    def get_thumb_path(self, file_id, thumbnail_frame, clear_cache=False):
+            # We didn't discover an image sequence
+            return None
+
+        # Found a sequence, ignore this path (no matter what the user answers)
+        # To avoid issues with overlapping/conflicting sets of files,
+        # we only attempt one image sequence match per directory
+        log.debug("Ignoring path for image sequence imports: {}".format(dirName))
+        self.ignore_image_sequence_paths.append(dirName)
+
+        log.info('Prompt user to import sequence starting from {}'.format(fileName))
+        if not get_app().window.promptImageSequence(fileName):
+            # User said no, don't import as a sequence
+            return None
+
+        # Yes, import image sequence
+        parameters = {
+            "folder_path": dirName,
+            "base_name": base_name,
+            "fixlen": fixlen,
+            "digits": digits,
+            "extension": extension
+        }
+        return parameters
+
+    def process_urls(self, qurl_list):
+        """Recursively process QUrls from a QDropEvent"""
+        import_quietly = False
+        media_paths = []
+
+        for uri in qurl_list:
+            filepath = uri.toLocalFile()
+            if not os.path.exists(filepath):
+                continue
+            if filepath.endswith(".osp") and os.path.isfile(filepath):
+                # Auto load project passed as argument
+                self.win.OpenProjectSignal.emit(filepath)
+                return True
+            if os.path.isdir(filepath):
+                import_quietly = True
+                log.info("Recursively importing {}".format(filepath))
+                try:
+                    for r, _, f in os.walk(filepath):
+                        media_paths.extend([
+                            os.path.join(r, p) for p in f
+                            ])
```

Comment:
```suggestion
                        media_paths.extend(
                            [os.path.join(r, p) for p in f])
```

Nice to be able to actually SEE it's a list comprehension.